source/mcpele/uniform_cubic_sampling.h

```diff
@@ -1,23 +1,35 @@
 #ifndef _MCPELE_UNIFORM_CUBIC_SAMPLING_H__
 #define _MCPELE_UNIFORM_CUBIC_SAMPLING_H__
 
+#include "pele/array.h"
+
 namespace mcpele {
     
 class UniformCubicSampling : public TakeStep {
 protected:
     std::mt19937_64 m_gen;
-    std::uniform_real_distribution<double> m_dist;
+    std::uniform_real_distribution<double> m_dist05;
+    pele::Array<double> m_boxvec;
+    bool m_cubic;
 public:
     virtual ~UniformCubicSampling() {}
-    UniformCubicSampling(const size_t seed=42, const double delta=1)
+    UniformCubicSampling(const size_t seed=42, const pele::Array<double> boxvec={2})
```

Comment:
what does this mean? `boxvec={2}`

Is it a default value of boxvec of length 1 with value 2?  Why would you want that?